src/aoc.rs

```diff
@@ -317,3 +320,103 @@ pub fn read(
     println!("\n{}", from_read(desc.as_bytes(), col_width));
     Ok(())
 }
+
+fn get_private_leaderboard_results(
+    args: &Args,
+    session: &str,
+    leaderboard: &str,
+    year: PuzzleYear,
+) -> AocResult<PrivateLeaderboard> {
+    debug!("🦌 Fetching private leaderboard {}", leaderboard);
+
+    let url = format!(
+        "https://adventofcode.com/{}/leaderboard/private/view/{}.json",
+        year, leaderboard
+    );
+
+    let leaderboard: PrivateLeaderboard =
+        build_client(session, "application/json")?
+            .get(&url)
+            .send()
+            .and_then(|response| response.error_for_status())
+            .and_then(|response| response.json())
+            .map_err(AocError::from)?;
+    Ok(leaderboard)
+}
+
+pub fn show_private_leaderboard_results(
+    args: &Args,
+    session: &str,
+    leaderboard: &str,
+) -> AocResult<()> {
+    let (year, day) = puzzle_year_day(args.year, args.day)?;
+    let leaderboard =
+        get_private_leaderboard_results(args, session, leaderboard, year)?;
+
+    let mut members: Vec<_> = leaderboard.members.values().collect();
+    members.sort_by_key(|m| Reverse(m.local_score));
+    members.iter().enumerate().for_each(|(idx, m)| {
+        let display_name = m
+            .name
+            .clone()
+            .unwrap_or(format!("anonymous user #{}", m.id));
+
+        let stars: String = (1..=25)
+            .map(|d| {
+                if d > day {
+                    ' '
+                } else {
+                    let stars = m.stars_per_day(d);
+                    match stars {
+                        2 => '★',
+                        1 => '☆',
+                        _ => '.',
+                    }
+                }
+            })
+            .collect();
+
+        let order = idx + 1;
+        println!("{}\t{}\t{}\t{}", order, m.local_score, stars, display_name);
+    });
+
+    Ok(())
+}
+
+#[derive(Deserialize)]
+struct PrivateLeaderboard {
+    owner_id: usize,
+    event: String,
+    members: HashMap<String, Member>,
+}
+
+#[derive(Deserialize)]
+struct Member {
```

Comment:
We could simplify this by implementing `TryFrom<&Value> for Member` so we only load the fields we need. We could even count the number of stars for each day then rather than when we're writing to the terminal. This would avoid all the `fields xxx are never read` clippy warnings.